src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -787,6 +801,7 @@ private async Task StartRequestAsync(WinHttpRequestState state)
                 EnsureSessionHandleExists(state);
 
                 SetEnableHttp2PlusClientCertificate(state.RequestMessage.RequestUri, state.RequestMessage.Version);
+                SetDisableHttp2StreamQueue();
```

Comment:
It looks like you're calling this for every request, is that correct?. If so, this is problematic as the event log will be flooded. This method sets a property on the session, not the request -- can this be moved into `EnsureSessionHandleExists` so it is only called a single time?

Reply:
I moved this call. But, there is also `SetEnableHttp2PlusClientCertificate` being called on each request that sets a session option as well. Do you think it should be moved too? (Note, the commits introduced `SetEnableHttp2PlusClientCertificate` have been already cherry-picked to .NET Core 3.1).

Reply:
Yes it looks like that should be moved too

Reply:
Got it. But, I think we need a separate PR for this because it might need to be backported to .NET Core 3.1.